File: languages/csharp_unity/Species.cs
using System;
using System.Collections.Generic;
[System.Serializable]
public class Species : BaseElement
{
    // Biology
    public string Appearance;
    public int? LifeSpan;
    public int? Weight;
    public List<string> NourishmentIds; = new List<string>();
    public List<string> ReproductionIds; = new List<string>();
    public List<string> AdaptationsIds; = new List<string>();
    // Psychology
    public string Instincts;
    public string Sociality;
    public string Temperament;
    public string Communication;
    public int? Aggression;
    public List<string> TraitsIds; = new List<string>();
    // World
    public string Role;
    public string ParentSpeciesId;
    public List<string> LocationsIds; = new List<string>();
    public List<string> ZonesIds; = new List<string>();
    public List<string> AffinitiesIds; = new List<string>();
}

File: languages/typescript/Species.ts
import { BaseElement } from './base_elements';

export interface Species extends BaseElement {
  // Biology
  appearance?: string | null;
  lifeSpan?: number | null;
  typicalWeight?: number | null;
  dietIds?: string[] | null;
  reproductionIds?: string[] | null;
  // Psychology
  instincts?: string | null;
  sociality?: string | null;
  temperament?: string | null;
  communication?: string | null;
  aggression?: number | null;
  traitsIds?: string[] | null;
  // World
  role?: string | null;
  parentSpeciesId?: string | null;
  locationsIds?: string[] | null;
  zonesIds?: string[] | null;
  affinitiesIds?: string[] | null;
}
